functions-and-scope.js

```diff
@@ -1,84 +1,145 @@
+    [v] Schrijf de stapjes om dit te kunnen checken eerst uit en vraag jezelf de volgende dingen af:
+    [v] * Hoe kan ik iedere waarde van de array checken op deze conditie?
+        1. maak een for loop.
+        2. check of getal hoger of gelijk is aan 8.
+        3. maak een variabelen "aantalCumLaude" aan die bij true mee telt
+    [v] * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
+        1. door naam index < parameter.lengt te gebruiken in de range van de for loop.
+    [v] * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+        1. maak een variabelen "aantalCumLaude" aan die bij true mee telt
+    [v] Log het antwoord in de terminal.
+
+        // ---- Verwachte uitkomst: 6
+*/
+
+function cumLaude(grades) {
+
+    let numberOfCumLaude = 0;
+
+    for (let i = 0; i < grades.length; i++) {
+        if (grades[i] >= 8) {
+            numberOfCumLaude++;
+        }
+    }
+    return numberOfCumLaude;
+}
+
+/*  1b: Omschrijven tot een herbruikbare functie
+    [v] Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+    [v] Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+    [v] Log het antwoord in de terminal.
+
+        ---- Verwachte uitkomsten:
+    [v] cumLaude(grades) geeft 6
+    [v] cumLaude([6, 4, 5]) geeft 0
+    [v] cumLaude([8, 9, 4, 6, 10]) geeft 3
+*/
+console.log("===================antwoordt opdracht 1=========================");
+console.log(cumLaude(grades));
+console.log(cumLaude([6, 4, 5]));
+console.log(cumLaude([8, 9, 4, 6, 10]));
+
+/* Opdracht 2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven
+    [v] De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is,
+    [v] Maar we beginnen met de grades array van hierboven.
+    [v] Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+    [v] * Hoe wordt een gemiddelde berekend?
+        1. alles bij elkaar op tellen en dan delen door het aantal cijfers
+    [v] * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+        1. variabele met alle cijfers op geteld
+        2. variabelen met het aantal indexen
+    [v] * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+        1. for loop met naam index < parameter.lengt te gebruiken
+    [v] Log het antwoord in de terminal.
+
+        // ---- Verwachte uitkomst: 6.642857142857143
+*/
+
+function averageGrade(grades) {
+
+    let totalAmountOfGrads = 0;
+    let totalNumbersOfGrads = 0;
+
+    for (let i = 0; i < grades.length; i++) {
+        totalAmountOfGrads = grades[i] + totalAmountOfGrads;
```

Comment:
Dit kun je versimpelen
```suggestion
        totalAmountOfGrads += grades[i];
```